File: 프로그래머스/3/12927. 야근 지수/야근 지수.py
def solution(n, works):
    answer = 0
    dic = {}
    for work in works:
        if work not in dic:
            dic[work] =1
        else : dic[work] +=1
        
    while n>0:
        max_key = max(dic)
        
        if max_key == 0:
            return 0
        
        if dic[max_key] < n:
            n -= dic[max_key]

            if max_key-1 not in dic:
                dic[max_key-1] =dic[max_key]
            else : dic[max_key-1] += dic[max_key]
            
            del dic[max_key]
            
        else:
            dic[max_key] -=n
            
            if max_key-1 not in dic:
                dic[max_key-1] = n
            else : dic[max_key-1] +=n
            
            n = 0
        
    for key in dic:
        answer += (key**2)*dic[key]
    return answer

File: 프로그래머스/3/12927. 야근 지수/야근 지수.js
function solution(n, works) {
    var answer = 0;
    works.sort((a,b)=>b-a)
    
    while (n>0){
        if (works[0] === 0) return 0;
        works[0]--;
        n--;
        i = 0;
        while (i < works.length-1 && works[i] < works[i+1]){
            [works[i],works[i+1]] = [works[i+1],works[i]];
            i++;
        }
    }
    
    works.map((value)=> answer += value**2)
    return answer;
}